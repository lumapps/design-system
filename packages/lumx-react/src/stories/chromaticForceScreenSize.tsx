/* eslint-disable */
import React from 'react';
import isChromatic from 'chromatic/isChromatic';

/** Story decorator used to force a minimum screen size when running in chromatic */
export const chromaticForceScreenSize = (story: any) =>
    isChromatic() ? <div style={{ minWidth: 1200, minHeight: 800 }}>{story()}</div> : story();
