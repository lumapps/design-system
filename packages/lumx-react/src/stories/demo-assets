../../../site-demo/static/demo-assets