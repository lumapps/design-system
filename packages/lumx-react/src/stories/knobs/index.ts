export * from './image';
export * from './lorem';
