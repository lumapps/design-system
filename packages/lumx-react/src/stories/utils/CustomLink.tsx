/* eslint-disable jsx-a11y/anchor-has-content,jsx-a11y/anchor-is-valid */
import React from 'react';

/**
 * Example custom link that can be used in linkAs props.
 */
export const CustomLink: any = ({ ...props }) => <a href="#" data-custom-link {...props} />;
