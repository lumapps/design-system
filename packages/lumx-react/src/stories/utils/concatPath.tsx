export type PathPart = string | number;
export type Path = PathPart | Array<PathPart>;

/**
 * Concat flatten object path
 *
 * @example concatPath('foo', 'bar') // => 'foo.bar'
 * @example concatPath(['foo', 0]) // => 'foo[0]'
 * @example concatPath('foo', 0, ['bar']) // => 'foo[0].bar'
 */
export const concatPath = (...prefix: Path[]) => {
    const [first, ...rest] = prefix.flat();
    return rest.reduce<string>((acc, part) => {
        if (typeof part === 'number') return `${acc}[${part}]`;
        return `${acc}.${part}`;
    }, String(first));
};
