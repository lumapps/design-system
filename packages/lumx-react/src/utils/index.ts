/**
 * WARNING: All modules exported here are exposed to NPM in '@lumx/react/utils'.
 */

export { ClickAwayProvider } from './ClickAwayProvider';
export { Portal, type PortalProps, type PortalInit, PortalProvider, type PortalProviderProps } from './Portal';
