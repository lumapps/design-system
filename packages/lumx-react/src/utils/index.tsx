export * from '@lumx/core/js/utils';

export * from './getRootClassName';

export * from './htmlDecode';

export * from './partitionMulti';

export * from './type';
