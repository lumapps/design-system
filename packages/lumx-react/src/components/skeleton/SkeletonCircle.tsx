import classNames from 'classnames';
import React, { forwardRef } from 'react';

import { GlobalSize, Theme } from '@lumx/react';
import { Comp, GenericProps, getRootClassName, handleBasicClasses } from '@lumx/react/utils';

/**
 * Defines the props of the component.
 */
export interface SkeletonCircleProps extends GenericProps {
    /** Size variant. */
    size: GlobalSize;
    /** Theme. */
    theme?: Theme;
}

const DEFAULT_PROPS: Partial<SkeletonCircleProps> = {
    theme: Theme.light,
};

/**
 * Component display name.
 */
const COMPONENT_NAME = 'SkeletonCircle';

/**
 * Component default class name and class prefix.
 */
const CLASSNAME = getRootClassName(COMPONENT_NAME);

/**
 * SkeletonCircle component.
 *
 * @param  props Component props.
 * @param  ref   Component ref.
 * @return React element.
 */
export const SkeletonCircle: Comp<SkeletonCircleProps, HTMLDivElement> = forwardRef((props, ref) => {
    const { className, size, theme, ...forwardedProps } = props;

    return (
        <div
            ref={ref}
            {...forwardedProps}
            className={classNames(className, handleBasicClasses({ prefix: CLASSNAME, size, theme }))}
        />
    );
});
SkeletonCircle.displayName = COMPONENT_NAME;
SkeletonCircle.defaultProps = DEFAULT_PROPS;
SkeletonCircle.className = CLASSNAME;
