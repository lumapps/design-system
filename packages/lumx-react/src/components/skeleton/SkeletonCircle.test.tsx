import React from 'react';
import { mount, shallow } from 'enzyme';
import 'jest-enzyme';
import { commonTestsSuite, itShouldRenderStories } from '@lumx/react/testing/utils';

import { SkeletonCircle, SkeletonCircleProps } from './SkeletonCircle';
import * as stories from './SkeletonCircle.stories';

const CLASSNAME = SkeletonCircle.className as string;

/**
 * Mounts the component and returns common DOM elements / data needed in multiple tests further down.
 */
const setup = (props: Partial<SkeletonCircleProps> = {}, shallowRendering = true) => {
    const renderer: any = shallowRendering ? shallow : mount;
    const wrapper: any = renderer(<SkeletonCircle {...(props as any)} />);
    return { props, wrapper };
};

describe(`<${SkeletonCircle.displayName}>`, () => {
    // 1. Test render via snapshot.
    describe('Snapshots and structure', () => {
        itShouldRenderStories(stories, SkeletonCircle);
    });

    // Common tests suite.
    commonTestsSuite(setup, { className: 'wrapper', prop: 'wrapper' }, { className: CLASSNAME });
});
