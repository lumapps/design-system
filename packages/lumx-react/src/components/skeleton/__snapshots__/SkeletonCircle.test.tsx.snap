// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkeletonCircle> Snapshots and structure should render story 'Circle' 1`] = `
Array [
  <div
    className="lumx-spacing-margin lumx-skeleton-circle lumx-skeleton-circle--size-xxs lumx-skeleton-circle--theme-light"
  />,
  <div
    className="lumx-spacing-margin lumx-skeleton-circle lumx-skeleton-circle--size-xs lumx-skeleton-circle--theme-light"
  />,
  <div
    className="lumx-spacing-margin lumx-skeleton-circle lumx-skeleton-circle--size-s lumx-skeleton-circle--theme-light"
  />,
  <div
    className="lumx-spacing-margin lumx-skeleton-circle lumx-skeleton-circle--size-m lumx-skeleton-circle--theme-light"
  />,
  <div
    className="lumx-spacing-margin lumx-skeleton-circle lumx-skeleton-circle--size-l lumx-skeleton-circle--theme-light"
  />,
  <div
    className="lumx-spacing-margin lumx-skeleton-circle lumx-skeleton-circle--size-xl lumx-skeleton-circle--theme-light"
  />,
  <div
    className="lumx-spacing-margin lumx-skeleton-circle lumx-skeleton-circle--size-xxl lumx-skeleton-circle--theme-light"
  />,
]
`;
